### ISSUE TITLE
Extract custom hooks from components

### ISSUE THREAD

**Issue by AlbertoPdRF** [AlbertoPdRF/tisn.app#226]
**Is your feature request related to a problem? Please describe.**

Components with a lot of logic (particularly custom hooks) are very hard to follow, especially for people who haven't created them. This also makes testing harder!

**Describe the solution you'd like**

I'd like all possible logic to be extracted from components. There's already a `logic` folder with a lot of functionality extracted, and the idea is to extract all custom hooks into their own folder too.

**Describe alternatives you've considered**

The only alternative is to keep everything as is, which is not ideal.

**Additional context**

None that I can think of at the moment!

**Comment by machadolucasvp** [comment]
would like to work on this, can you assign to me @AlbertoPdRF ?
